src/service/device.js

```diff
@@ -338,7 +338,10 @@ var Device = GObject.registerClass({
             let packet = null;
 
             while ((packet = await this.channel.readPacket())) {
-                debug(packet, this.name);
+                if (globalThis.debugDetail)
+                    debug(packet, this.name);
+                else
+                    debug(packet.type, this.name);
```

Comment:
I think this check would be better only done once (ie. swapping the callback when the setting changes), but this is good enough for now. It can always be changed later.

Reply:
Hmm, that could be tricky because only those two debug calls should be abbreviated. I suppose I could add a separate debugging call that only those two functions use, which would either be the same callback as `debug()` (when details are enabled), or an abbreviating version (when they aren't). That makes sense, and it's easy enough. (Then I won't need to expose `debugDetail` via `globalThis`, either.)

Reply:
@andyholmes  Sorted, hopefully, though the changes are significant enough that I wouldn't say no to a second look.

Reply:
@andyholmes Whoops, hopefully you hadn't looked (again) yet, because the code wasn't done. (Forgot to _save_ in VSCode. Argh.) **Now** it is.